src/webgpu/shader/execution/expression/call/builtin/bitcast.spec.ts

```diff
@@ -158,13 +319,320 @@ function bitcastI32ToF32Comparator(i: number): Comparator {
  */
 function bitcastU32ToF32Comparator(u: number): Comparator {
   const f: number = reinterpretU32AsF32(u);
-  if (!isFinite(f)) return anyU32;
+  if (!isFiniteF32(f)) return anyU32;
   // Positive or negative zero bit pattern map to any zero.
   if (f32ZerosInU32.includes(u)) return anyOf(...f32ZerosInF32.map(f32));
   const acceptable: number[] = [f, ...(isSubnormalNumberF32(f) ? f32ZerosInF32 : [])];
   return anyOf(...acceptable.map(f32));
 }
 
+/**
+ * @returns an array of expected f16 FPInterval for the given bitcasted f16 value, which may be
+ * subnormal, Inf, or NaN. Test cases that bitcasted to vector of f16 use this function to get
+ * per-element expectation and build vector expectation using cartesianProduct.
+ */
+function generateF16ExpectationIntervals(bitcastedF16Value: number): FPInterval[] {
+  // If the bitcasted f16 value is inf or nan, the result is unbounded
+  if (!isFiniteF16(bitcastedF16Value)) {
+    return [f16UnboundedInterval];
+  }
+  // If the casted f16 value is +/-0.0, the result can be one of both. Note that in JS -0.0 === 0.0.
+  if (bitcastedF16Value === 0.0) {
+    return [f16ZerosInterval];
+  }
+  const exactInterval = FP.f16.toInterval(bitcastedF16Value);
+  // If the casted f16 value is subnormal, it also may be flushed to +/-0.0.
+  return [exactInterval, ...(isSubnormalNumberF16(bitcastedF16Value) ? [f16ZerosInterval] : [])];
```

Comment:
Nice catch in handling the case that a subnormal value can map to a zero of the opposite sign.